src/plugins/columnSorting/columnSorting.js

```diff
@@ -114,97 +135,66 @@ class ColumnSorting extends BasePlugin {
    * Disable plugin for this Handsontable instance.
    */
   disablePlugin() {
-    this.hot.sort = void 0;
     super.disablePlugin();
   }
 
   /**
-   * afterUpdateSettings callback.
-   *
-   * @private
+   * Destroy plugin instance.
    */
-  onAfterUpdateSettings() {
-    this.sortBySettings();
-  }
+  destroy() {
+    this.rowsMapper.destroy();
 
-  sortBySettings() {
-    let sortingSettings = this.hot.getSettings().columnSorting;
-    let loadedSortingState = this.loadSortingState();
-    let sortingColumn;
-    let sortingOrder;
-
-    if (typeof loadedSortingState === 'undefined') {
-      sortingColumn = sortingSettings.column;
-      sortingOrder = sortingSettings.sortOrder;
-    } else {
-      sortingColumn = loadedSortingState.sortColumn;
-      sortingOrder = loadedSortingState.sortOrder;
-    }
-    if (typeof sortingColumn === 'number') {
-      this.lastSortedColumn = sortingColumn;
-      this.sortByColumn(sortingColumn, sortingOrder);
-    }
+    super.destroy();
   }
 
   /**
-   * Set sorted column and order info
+   * Sorting the table by chosen column and order.
    *
-   * @param {number} col Sorted visual column index.
-   * @param {boolean|undefined} order Sorting order (`true` for ascending, `false` for descending).
+   * @param {Number} column Visual column index.
+   * @param {undefined|String} order Sorting order (`asc` for ascending, `desc` for descending and `none` for initial state).
    */
-  setSortingColumn(col, order) {
-    if (typeof col == 'undefined') {
-      this.hot.sortColumn = void 0;
-      this.hot.sortOrder = void 0;
+  sort(column, order) {
+    this.setSortingColumn(column, order);
 
+    if (isUndefined(this.sortColumn)) {
       return;
-    } else if (this.hot.sortColumn === col && typeof order == 'undefined') {
-      if (this.hot.sortOrder === false) {
-        this.hot.sortOrder = void 0;
-      } else {
-        this.hot.sortOrder = !this.hot.sortOrder;
-      }
-
-    } else {
-      this.hot.sortOrder = typeof order === 'undefined' ? true : order;
     }
 
-    this.hot.sortColumn = col;
-  }
-
-  sortByColumn(col, order) {
-    this.setSortingColumn(col, order);
-
-    if (typeof this.hot.sortColumn == 'undefined') {
-      return;
-    }
-
-    let allowSorting = this.hot.runHooks('beforeColumnSort', this.hot.sortColumn, this.hot.sortOrder);
+    const allowSorting = this.hot.runHooks('beforeColumnSort', this.sortColumn, this.sortOrder);
```

Comment:
In this case, if `beforeColumnSort` returns false, then rest this should be the end of sorting action.